pennylane/beta/plugins/default_tensor.py

```diff
@@ -428,38 +422,167 @@ def ev(self, obs_nodes, wires, contraction_method="auto"):
             )
         return self._real(expval)
 
-    def _contract_to_ket(self, contraction_method="auto"):
-        """Contract the nodes which represent the state preparation and gate applications to get the pre-measurement state.
+    def _ev_exact(self, obs_nodes, wires):
+        r"""Expectation value of observables on specified wires using an exact representation.
+
+         Args:
+            obs_nodes (Sequence[tn.Node]): the observables as TensorNetwork Nodes
+            wires (Sequence[Sequence[int]]): measured subsystems for each observable
 
-        The contracted tensor is stored in the ``_nodes`` dictionary under the key ``"contracted_state"``.
+         Returns:
+            complex: expectation value :math:`\expect{A} = \bra{\psi}A\ket{\psi}`
+        """
+        self._contract_premeasurement_network()
+        ket = self._nodes["contracted_state"]
+        bra = tn.conj(ket, name="Bra")
```

Comment:
I'm slightly curious if the proposal in [ADR69](https://github.com/XanaduAI/software-docs/blob/master/src/research/0069-tensor-contractions.rst#expectation-values-via-decomposition) would make any difference with respect to memory/speed. I haven't tested it myself, but not 100% sure if it's worth the additional effort for now.

Reply:
leaving open for future consideration